deepmd/pd/train/training.py

```
@@ -0,0 +1,1249 @@
+        fout = (
+            open(
+                self.disp_file,
+                mode="w" if not self.restart_training else "a",
+                buffering=1,
+            )
+            if self.rank == 0
+            else None
+        )  # line buffered
+        if SAMPLER_RECORD:
+            record_file = f"Sample_rank_{self.rank}.txt"
+            fout1 = open(record_file, mode="w", buffering=1)
```

Ruff (0.8.0)

Use a context manager for opening files

(SIM115)